### PR TITLE
add volume internal block

### PR DESCRIPTION
**Pull request by losynix**
Volume internal block:

exec: "YABAR_VOLUME";
interval: 1;
internal-option1 : "default"; # device
internal-option2 : "Master 0"; # Mixer index (separated by a space)
internal-option3 : " "; # characters to display when sound is on or off (separated by a space)
internal-suffix: "%";

### CODE REVIEW

**Comment by erebe** [comment]
\o/ I will be able to remove my patch of your pull request

File: doc/yabar.1 [hunk 1]
```diff
@@ -517,6 +517,21 @@ internal-option2: "        ";
...
+Volume: It uses ALSA to display sound volume in percentage. Example:
+.PP
+.RS
+
+.nf
+exec: "YABAR_VOLUME";
+interval: 1;
+internal-option1 : "default"; # device
+internal-option2 : "Master 0"; # Mixer index (separated by a space)
+internal-option3 : " "; # characters to display when sound is on or off (separated by a space)
```
Comment (NBonaparte) [inline doc/yabar.1]: We recently changed the delimiter to a semicolon, could you change it here too? (Just put one between the two characters)

File: examples/example.config [hunk 2]
```diff
@@ -40,6 +40,17 @@ topbar:{
 		internal-prefix: " ";
 		internal-option1: "%a %d %b, %I:%M";
 	}
+	ya_volume:{
+		exec: "YABAR_VOLUME";
+		align: "right";
+		interval: 1;
+		internal-option1 : "default"; # device
+		internal-option2 : "Master 0"; # Mixer index (separated by a space)
+		internal-option3 : " "; # characters to display when sound is on or off (separated by a space)
```
Comment (NBonaparte) [inline examples/example.config]: Same here.

File: src/intern_blks/ya_intern.c [hunk 3]
```diff
@@ -563,6 +565,100 @@ void ya_int_battery(ya_block_t *blk) {
...
+		goto ya_volume_error;
+	}
+	if( blk->internal->option[1] ) {
+		sscanf(blk->internal->option[1], "%s %d", mixer_name, &mixer_index);
+	} else {
+		fprintf(stderr, "YABAR_VOLUME : internal-option2 is mandatory\n");
+		goto ya_volume_error;
+	}
+	if( blk->internal->option[2] ) {
+		sscanf(blk->internal->option[2], "%s %s", on, off);
```
Comment (NBonaparte) [inline src/intern_blks/ya_intern.c]: And here.

**Comment by jacksgt** [comment]
Please rebase your branch on top of master, since I have just merged #1 from @erebe

**Comment by jacksgt** [comment]
Sorry, but something is not quite right with your pull request (as you can see it says "This branch is 4 commits ahead, 3 commits behind yabar-dev:master.", and for some reason not even the Travis builds are running).
Could you please check that?

**Comment by losynix** [comment]
Yes sorry, trying to fix this

**Comment by jacksgt** [comment]
Also, I think this block should contain ALSA somewhere in its name, since this will not work for PulseAudio setups. Maybe "YABAR_ALSA_VOLUME", "YABAR_ALSA_VOL" or just "YABAR_ALSA" ...

**Comment by losynix** [comment]
I'm actually using PulseAudio (which uses Alsa's channels) and it's working well but yes maybe one would want to have multiple volume blocks for every pulse audio sinks. I don't know what to do about the conflicts, any idea ?

**Comment by losynix** [comment]
Should be ok now, I messed up the first rebase

**Comment by jacksgt** [comment]
> I'm actually using PulseAudio (which uses Alsa's channels) and it's working well but yes maybe one would want to have multiple volume blocks for every pulse audio sinks.
I'm curious now, when you change your PA Volume (e.g. via pavucontrol), does this block display that?

**Comment by erebe** [comment]
I have pulseaudio and pavucontrol and it does work as expected on my system.

```
	volume:{
		exec: "YABAR_VOLUME";
		align: "right";
		interval: 1;
		internal-option1 : "default"; # device
		internal-option2 : "Master 0"; # Mixer index (separated by a space)
		internal-option3 : " "; # characters to display when sound is on or off (separated by a space)
		internal-suffix: "%";
		underline-color-rgb: 0xf5d63d;
        background-color-argb: 0xD0212121;
        command-button1: "pavucontrol";
        command-button3: "pactl set-sink-mute 0 toggle";
        command-button4: "pactl set-sink-volume 0 +10%";
        command-button5: "pactl set-sink-volume 0 -10%";
	}
```

**Comment by losynix** [comment]
Yes indeed, the "Built-in Audio Analog Stereo" under the "Output Devices" tab of pavucontrol (default sink of pulseaudio) appears to be linked to the Master:0 mixer of Alsa. On my machine at least but it is configurable.

**Comment by jacksgt** [comment]
Huh, that's nice. I knew PA works on top of Alsa, but I always thought they use independent volume levels.
Thanks for that info and for your contribution!

**Comment by jacksgt** [comment]
Added documentation to wiki:
https://github.com/yabar-dev/yabar/wiki/Internal-Blocks#yabar_volume

**Comment by losynix** [comment]
Thank you ! For the documentation and for resurrecting this project !

**Pull request merged**